tracing-test-utils/src/main/java/com/palantir/tracing/TestTracingExtension.java

```diff
@@ -55,7 +55,9 @@ public void beforeTestExecution(ExtensionContext context) {
     @Override
     public void afterTestExecution(ExtensionContext context) throws Exception {
         String name = testName(context);
-        Tracer.unsubscribe(context.getUniqueId());
+        if (!(Tracer.unsubscribe(context.getUniqueId()) instanceof TestTracingSubscriber subscriber)) {
+            throw new SafeIllegalStateException("Expecting TestTracingSubscriber");
+        }
```

Comment:
This seems unnecessary

Reply:
Will need to cast it to have the getAllSpans method. I can do a direct cast if that's more preferable?

Reply:
Not obvious from this PR diff but `subscriber` is used below.